release-notes/schemas/dotnet-os-packages.json

```diff
@@ -0,0 +1,190 @@
+{
+  "description": "The set of packages required by a given product version for a set of distros.",
+  "type": [
+    "object",
+    "null"
```

Comment:
What does this `null` stuff even mean for the type? Does that mean its nullable? How would that be represented in the JSON? Is it applicable in all cases?

Reply:
I don't know how that works. @eiriktsarpalis  can likely explain.

Reply:
It means that the underlying .NET type is a reference type and reflects the fact that the serializer will accept `null` JSON values when deserializing. There was a long debate as to whether we should be exposing `null` in the schema and ultimately we chose correctness over conciseness.

Reply:
Requiredness and nullability are orthogonal traits. This is true both in JSON schema and in C# semantics where you can have `required` fields that are nullable.

Reply:
Conversely, you can have non-nullable parameters that are optional, e.g.
```C#
record MyRecord(string value = "default");
```

Reply:
I get it now. It's not exactly intuitive, but I guess it makes sense. In any case, your suggested pattern works. Thanks!

Reply:
This PR doesn't include the code that was used to generate this schema, correct?

Reply:
It does not. I haven't figured out the long-term home for these tools.

Schema generation: https://github.com/richlander/distroessed/blob/linux-packages/src/GenerateJsonSchemas/Program.cs

OM: https://github.com/richlander/distroessed/blob/linux-packages/src/DotnetRelease/PackageRecords.cs